lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -431,18 +455,56 @@ def create_coefficients_cube(
             coeff_names, long_name="coefficient_name", units="no_unit")
         dim_coords_and_dims = [(coefficient_index, 0)]
         aux_coords_and_dims = [(coefficient_name, 0)]
-        for coord_name in (
-                ["time", "forecast_period", "forecast_reference_time"]):
-            try:
-                aux_coords_and_dims.append(
-                    (current_forecast.coord(coord_name), None))
-            except CoordinateNotFoundError:
-                pass
-        attributes = {"diagnostic_standard_name": current_forecast.name()}
-        for attribute in current_forecast.attributes.keys():
+
+        # Create a forecast_reference_time coordinate.
+        frt_point = cycletime_to_datetime(self.current_cycle)
+        try:
+
+            frt_coord = (
+                historic_forecast.coord("forecast_reference_time").copy(
+                    datetime_to_iris_time(frt_point, time_units="seconds")))
+        except CoordinateNotFoundError:
+            pass
+        else:
+            aux_coords_and_dims.append((frt_coord, None))
+
+        # Create forecast period and time coordinates.
+        try:
+            fp_point = (
+                np.unique(historic_forecast.coord("forecast_period").points))
```

Comment:
The previous comments here suggested this needed to be converted to seconds. Why have you chosen not to do this?

Reply:
This is just for the purpose of supporting inputs with different units for the forecast_period. With the current structuring of the code, the conversion of forecast_period to seconds only occurs where it is required i.e. to support the construction of the `timedelta` object.